pkg/registry/core/pod/storage/eviction.go

```diff
@@ -303,11 +303,11 @@ func resourceVersionIsUnset(options *metav1.DeleteOptions) bool {
 }
 
 func createTooManyRequestsError(name string) error {
-	// TODO(mml): Add a Retry-After header.  Once there are time-based
+	// TODO(mml): Once there are time-based
 	// budgets, we can sometimes compute a sensible suggested value.  But
-	// even without that, we can give a suggestion (10 minutes?) that
+	// even without that, we can give a suggestion (10 minutes) that
 	// prevents well-behaved clients from hammering us.
-	err := errors.NewTooManyRequests("Cannot evict pod as it would violate the pod's disruption budget.", 0)
+	err := errors.NewTooManyRequests("Cannot evict pod as it would violate the pod's disruption budget.", 600)
```

Comment:
Looking into the code, it seems that this path isn't really exercised on overload, but rather in two options:
(a) on conflict to force the client to retry:
https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/core/pod/storage/eviction.go#L262-L265
- if we're lagging behind the current version:
https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/core/pod/storage/eviction.go#L319

For (a) - returning 429 seems strange to me - what's the purpose of hiding "Confict" here? Users should retry conflict too? I think returning 429 is very misleading in this case.


For (b) - returning 429 is also misleading. I'm not sure what would be the option though - maybe 425 (StatusTooEarly):
https://golang.org/src/net/http/status.go

@liggitt - WDYT?

Reply:
The server already retries on conflict. This is surfaced to the client if that retry was exhausted.

If the client submits an eviction without any preconditions, it isn't really expecting a conflict response.

A 429 doesn't seem terrible here... it means something is overloading the server with PDB updates... either pod churn, or eviction churn.

Reply:
that said, a 10-minute backoff seems excessive to me